calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -162,15 +162,26 @@ func (rh resourceHelper) Apply(ctx context.Context, client client.Interface, res
 	// Try to create the resource first.
 	ro, err := rh.Create(ctx, client, resource)
 
-	// Fall back to an Update if the resource already exists, or the datastore does not support
-	// create operations for that resource.
 	switch err.(type) {
-	case cerrors.ErrorResourceAlreadyExists, cerrors.ErrorOperationNotSupported:
+	case cerrors.ErrorResourceAlreadyExists:
 		// Insert the original ResourceVersion back into the object before trying the Update.
 		resource.(ResourceObject).GetObjectMeta().SetResourceVersion(originalRV)
 
-		// Try updating if the resource already exists.
+		// Fall back to an Update if the resource already exists.
 		return rh.Update(ctx, client, resource)
+
+	case cerrors.ErrorOperationNotSupported:
+		// Insert the original ResourceVersion back into the object before trying the Update.
+		resource.(ResourceObject).GetObjectMeta().SetResourceVersion(originalRV)
+
+		// Try updating if the resource already exists.
+		_, nerr := rh.Update(ctx, client, resource)
+		if _, dne := nerr.(cerrors.ErrorResourceDoesNotExist); dne {
```

Comment:
I feel like this is wrong.

Take the kdd Node case as an example:
- Create node fails with unsupported
- Update node fails with "node does not exist"

The error the use should receive is "Node does not exist", not "operation unsupported"

Reply:
I think the UX you've outlined contradicts the one outlined here:

>If calicoctl were to do an "apply" action on a ClusterInformation resource, it will get the standard error that the resource is "not found" (since it will try an update). We should make sure that this error message gets switched to give out a "not supported" error messaged instead to make it clearer that you should not be updating a ClusterInformation resource.

Reply:
I don't see how.

For ClusterInformation, any operation should return "Not supported"

For nodes, "Create" should return not supported, but "Apply" and "Update" should not return "Not supported"